Add MPR121 support

mvturnho opened the pull request:
## Description:
Add MPR121 support

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

mvturnho committed:
solved issues image size, tabs to spaces, file location

mvturnho committed:
image name was incorrect

mvturnho committed:
image for index was missing

OttoWinter commented on components/binary_sensor/mpr121.rst:
```diff
@@ -0,0 +1,60 @@
+- **name** (**Optional**, string): The name for the binary sensor.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **channel** (*Required*, integer): The channel number at the mpr121 the touchkey is connected to.
+- All other options from :ref:`Binary Sensor <config-binary_sensor>` and :ref:`MQTT Component <config-mqtt-component>`.
```
This line has changed since the 1.11 update, can you copy the new way this line looks from the other binary sensor files?

mvturnho replied:
We need to remove the two parts `Base Conficuration` and `Binary Sensor Configuration` and only have the `Configuration variables`?

OttoWinter replied:
No I mean like this (see for example gpio binary sensor docs).

```suggestion
- All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```

OttoWinter commented on components/binary_sensor/mpr121.rst:
```diff
@@ -0,0 +1,60 @@
+`Adafruit`_) Capacitive Touch Sensor with esphomelib. The :ref:`I²C <i2c>` is
+required to be set up in your configuration for this sensor to work.
+
+.. figure:: images/mpr121.png
```
If the two images are exact duplicates you can also just remove the one in `binary_sensor` and use `/images/mpr121.png` here

mvturnho committed:
Merge pull request #1 from esphome/next

update fork to branch next

mvturnho committed:
remove duplicate image

OttoWinter committed:
Update images

- Convert main index image from .png to .jpg
- Use full picture on component page
- Add adafruit attribution

OttoWinter committed:
Remove MQTT ref

mvturnho committed:
fixed configuration rows

mvturnho committed:
Merge branch 'mpr121' of https://github.com/mvturnho/esphome-docs.git into mpr121

OttoWinter reviewed (approved):
👍
